### PR TITLE
maintenance: add auto-updating of npm packages

### PR DESCRIPTION
**Pull request by stephenplusplus**
This script will update our sub-modules to the latest compatible dependencies when run:

``` sh
$ npm run update-deps
```

Also included in the PR are the modified package.json's after running.

### CODE REVIEW

File: packages/dns/package.json [hunk 1]
```diff
@@ -52,7 +52,7 @@
   "dependencies": {
     "@google-cloud/common": "^0.5.0",
     "arrify": "^1.0.0",
-    "dns-zonefile": "0.1.18",
+    "dns-zonefile": "^0.1.18",
```
Comment (stephenplusplus) [inline packages/dns/package.json]: @callmehiphop We intentionally wanted 0.1.18 and not any updates, didn't we? I can exclude this if so.

Comment (callmehiphop) [inline packages/dns/package.json]: I think that's right, but I'd have to search for the reasoning behind that.

Comment (stephenplusplus) [inline packages/dns/package.json]: Added it to a blacklist anyway.

File: scripts/update-deps.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+var async = require('async');
+var david = require('david');
```
Comment (callmehiphop) [inline scripts/update-deps.js]: this is my new favorite package

Comment (stephenplusplus) [inline scripts/update-deps.js]: 💃

**Comment by callmehiphop** [comment]
So what's the long term plan for this? Run it once in a while to see if we have updates..? I think it would be great it we could tie this into CI somewhere and have it just run weekly or so..?

**Comment by stephenplusplus** [comment]
That's interesting. My plan was to simply run it manually before publishing.

**Pull request merged**